functests/xt_u32/xt_u32.go

```diff
@@ -0,0 +1,223 @@
+package xt_u32
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	igntypes "github.com/coreos/ignition/config/v2_2/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/execute"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/images"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/namespaces"
+	utilNodes "github.com/openshift-kni/cnf-features-deploy/functests/utils/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const hostnameLabel = "kubernetes.io/hostname"
+const TestNamespace = "test-xt-u32"
+
+const xt_u32LoadPath = "/etc/modules-load.d/xt_u32-load.conf"
+
+var (
+	xt_u32NodeSelector string
+	hasNonCnfWorkers   bool
+)
+
+func init() {
+	roleWorkerCNF := os.Getenv("ROLE_WORKER_CNF")
+	if roleWorkerCNF != "" {
+		xt_u32NodeSelector = fmt.Sprintf("node-role.kubernetes.io/%s=", roleWorkerCNF)
+	}
+
+	hasNonCnfWorkers = true
+	if os.Getenv("XT_U32_TEST_HAS_NON_CNF_WORKERS") == "false" {
+		hasNonCnfWorkers = false
+	}
+}
+
+var _ = Describe("xt_u32", func() {
+	var testNode string
+
+	execute.BeforeAll(func() {
+		err := namespaces.Create(TestNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = namespaces.Clean(TestNamespace, "xt-u32", client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		if xt_u32NodeSelector == "" {
+			xt_u32NodeSelector, err = findXT_U32NodeSelector()
```

Comment:
not sure, but shouldnt there be some distinction between discovery mode (and non-discovery mode)?
ie create the MC if needed?
@fedepaol

Reply:
The MC creation is intended to be the "installation of the feature". That's the approach we took with SCTP.
With that, it does not make much distinction between discovery and non discovery, apart from skipping or failing.
@sabinaaledort this reminds me that it would make sense to extend the validation suite (https://github.com/openshift-kni/cnf-features-deploy/blob/73e520f860d35a62e34c1e770a383e29b9c45dc7/validationsuite/cluster/validation.go#L171 ) the same way we did for SCTP